rtsp.go

```diff
@@ -494,16 +484,20 @@ func (rc *rtspCamera) initH264(session *description.Session) (err error) {
 		return fmt.Errorf("creating H264 raw decoder: %w", err)
 	}
 
-	if rc.vs != nil {
-		rc.logger.Info("video-store is not nil, attempting to call InitH264")
-		if len(f.SPS) != 0 && len(f.PPS) != 0 {
-			if err := rc.vs.InitH264(f.SPS, f.PPS); err != nil {
-				rc.logger.Errorf("videostore.InitH264 error: %s", err.Error())
-				return err
-			}
-			rc.logger.Info("videostore.InitH264 succeeded")
-		} else {
-			rc.logger.Warnf("sps or pps not present so video-store is not initializes sps len: %d, pps len: %d", len(f.SPS), len(f.PPS))
+	if rc.vsConfig != nil {
+		rc.logger.Info("creating video-store from config")
+		config := *rc.vsConfig
+		config.Type = videostore.SourceTypeH264RTPPacket
+		vs, err := videostore.NewRTPVideoStore(config, rc.logger)
+		if err != nil {
+			return err
+		}
+		rc.videoStoreMuxer = &videoStoreMuxer{
+			SourceType: config.Type,
+			vs:         vs,
+			sps:        f.SPS,
+			pps:        f.PPS,
+			logger:     rc.logger,
```

Comment:
This change makes it so that h264 and h265 video-stores don't depend on the initial sps, pps (and in the caes of h265 vps sent in the sdp). If the camera sends these only in the rtsp stream, video-store will still work.

We should update the storeImage code path to also be resillient to those parameters not being sent in the SDP as a follow up.